File: app/src/main/java/com/kite/mnemoai/ui/reciteword/ReciteWordUIState.java
package com.kite.mnemoai.ui.reciteword;

import com.kite.mnemoai.data.model.WordDetailInfo;
import com.kite.mnemoai.ui.model.LoadingState;

import java.util.List;

public class ReciteWordUIState {
    private List<WordDetailInfo> reciteWordItemStatusOrder;
    private ReciteStage reciteStage;
    private int currentProgress;
    private int totalProgress;
    private boolean isShowNext;
    private boolean isShowTranslation;
    private boolean isShowDetail;
    private LoadingState aiMnemonicLoadingState;

    public ReciteWordUIState(ReciteStage reciteStage,
                             List<WordDetailInfo> reciteWordItemStatusOrder,
                             int totalProgress,
                             int currentProgress,
                             boolean isShowNext,
                             boolean isShowTranslation,
                             boolean isShowDetail,
                             LoadingState aiMnemonicLoadingState) {
        this.reciteStage = reciteStage;
        this.reciteWordItemStatusOrder = reciteWordItemStatusOrder;
        this.totalProgress = totalProgress;
        this.currentProgress = currentProgress;
        this.isShowNext = isShowNext;
        this.isShowTranslation = isShowTranslation;
        this.isShowDetail = isShowDetail;
        this.aiMnemonicLoadingState = aiMnemonicLoadingState;
    }

    public List<WordDetailInfo> getReciteWordItemStatusOrder() {
        return reciteWordItemStatusOrder;
    }

    public void setReciteWordItemStatusOrder(List<WordDetailInfo> reciteWordItemStatusOrder) {
        this.reciteWordItemStatusOrder = reciteWordItemStatusOrder;
    }

    public boolean isShowNext() {
        return isShowNext;
    }

    public void setShowNext(boolean showNext) {
        this.isShowNext = showNext;
    }

    public ReciteStage getReciteStage() {
        return reciteStage;
    }

    public int getCurrentProgress() {
        return currentProgress;
    }

    public int getTotalProgress() {
        return totalProgress;
    }

    public LoadingState getAiMnemonicLoadingState() {
        return aiMnemonicLoadingState;
    }

    public boolean isShowDetail() {
        return isShowDetail;
    }

    public boolean isShowTranslation() {
        return isShowTranslation;
    }
}

File: feature/reciteword/src/main/java/com/kite/mnemoai/reciteword/ReciteWordUIState.kt
package com.kite.mnemoai.reciteword

import com.kite.mnemoai.model.Result
import com.kite.mnemoai.model.word.WordDetail

data class ReciteWordUIState(
    val reciteStage: ReciteStage?,
    val reciteWordItemStatusOrder: MutableList<WordDetail>?,
    val totalProgress: Int,
    val currentProgress: Int,
    val isShowTranslation: Boolean,
    val isShowDetail: Boolean,
    val aiMnemonicLoadingState: Result<String>?,
    /**
     * 单调递增的版本号：用于绕过 StateFlow 的 equals 去重，
     * 保证列表被重排回原序等"内容未变但 UI 需要刷新"的场景仍会发射。
     */
    val revision: Long = 0,
)
